lms/djangoapps/certificates/generation.py

```diff
@@ -59,14 +59,18 @@ def _generate_certificate(user, course_key):
     """
     Generate a certificate for this user, in this course run.
     """
+    # Retrieve the existing certificate for the learner if it exists
+    existing_certificate = GeneratedCertificate.certificate_for_student(user, course_key)
+
     profile = UserProfile.objects.get(user=user)
     profile_name = profile.name
 
     course = modulestore().get_course(course_key, depth=0)
     course_grade = CourseGradeFactory().read(user, course)
     enrollment_mode, __ = CourseEnrollment.enrollment_mode_for_user(user, course_key)
-    key = make_hashkey(random.random())
```

Comment:
I pulled the generation of new `key` values since I was touching this file. This goes back to MB-976 and our Spike on "what is `key` and `verify_uuid`?".

As far as I can tell, the `key` field was only used for PDF certificate generation. If we prefer to split this out, or just keep it in until we can remove the PDF certificate code, i'm totally okay with that.